### ISSUE TITLE
Store stopwords in corpus object

### ISSUE THREAD

**Issue by JaimieMurdock** [inpho/vsm#114]
Store stopwords in corpus object, rather than just eliminating them.

**Comment by rrose1** [comment]
In a much earlier version of vsm, we had this functionality. There was a `MaskedCorpus` class built on numpy's masked array class. The latter is not as robust as we needed it to be; this was one reason `MaskedCorpus` was deprecated. Another reason is that it makes training functions more complicated (it's really better to leave reindexing tasks out of training, to the extent possible).

My thought at the time was that the Corpus class should lack a lot of the functionality you might expect from a class so-named (as compared, for example, to a comparable nltk class); it should have just what a trainer would need. (This discussion relates too to the discussion about "context_type".) Since a user would want a "Corpus" object with a lot of functionality, I was considering having a class play the role of a "RawCorpus" which would have all of that functionality (and data and metadata). "RawCorpus" would have a method, for example, which would return a flat, stopped stripped-down `Corpus` object ready for training on. It might be more sensible to call this "RawCorpus" class `Corpus` and the training data `CorpusTrainingData` or some such thing.

However you decide to proceed, my main two suggestions would be (a) to try to keep the bookkeeping tasks out of the training functions proper (ideally, the algorithms should be easy-to-read in the training code and the implementation should be efficient) and (b) to take into careful consideration data storage and management platforms you'll be targeting in the future (for example, h5py).
